Usage Guide Documentation (#2675)
hikari-py/hikari: docs/guide/environment/index.md

# Environment

> Since `hikari` runs on modern, non-deprecated versions of Python, we first need to set up a proper development environment. This guide walks you through installing Python, `hikari`, and a suitable code editor.

Reviewer: ```suggestion
Since `hikari` works best on modern versions of Python, we first need to set up a proper development environment. This guide walks you through installing Python, `hikari`, and a suitable code editor.
```
I don't feel the `non-deprecated` part needs to be said

## Installing Python

It is strongly recommended to install the **latest stable version of Python**, as it includes performance improvements, bug fixes, and new language features.

=== "Windows"

    1. Visit Python's [official download page](https://www.python.org/downloads/)
    2. Select the latest release and scroll down to the **Files** section
        ![Python Files](../../assets/guide/environment/3.10.11-files.png "Python Files")

>     3. Download the **Windows installer (XX-bit)**

Reviewer: I want to point out (and TIL) that there is an alternative way to install Python on Windows (other than the Microsoft Store) in the form of the `.msix`-based [Python install manager](https://www.python.org/downloads/release/pymanager-263/). In their view this install manager is the official "modern" way to install Python while the older MSI-based installers are deprecated and will stop being released with Python 3.16 (which is just over a year away).
It might be beneficial to reword this section to mention the new installer and walk the user through that instead.

Reviewer: Also would add the warning to avoid the MS Store version like the plague

Reviewer 2: yeah never install from the MS store

Author: Sounds good to me.

        - Windows 11 is always **64-bit**

>         - Older versions may support both 32-bit and 64-bit

Reviewer: We could simplify this language by just directing the user to the 64-bit install. I haven't seen a 32-bit machine in well over a decade, and any user running 32-bit (if they're still alive) is probably well aware they're on a 32-bit machine

Author: That would make sense, I agree.

    4. Run the installer and **leave all default options enabled**
        - Ensure **Add Python to PATH** is checked
        - Ensure **pip** is selected for installation
    5. Verify Python's installation:
        ```powershell
        python --version
        ```

>         ![Verify Python](../../assets/guide/environment/verify-python.png "Verify Python")

Reviewer 2: This is a nitpick, you mentioned that people should select the latest version and your screenshot shows py 3.10 which is very near its EOL

Author: I only did it as it's the minimum version supported by hikari. I think I mentioned earlier in the guide that it's recommended to install the latest, but the guide will be using 3.10.11. I am unsure, however.

    6. Verify `pip`'s installation:
        ```powershell
        pip --version
        ```

        ![Verify pip](../../assets/guide/environment/verify-pip.png "Verify pip")

        If `pip` is not recognized, reinstall Python and make sure the **pip** option is **enabled**.

=== "macOS"

    1. Visit Python's [official download page](https://www.python.org/downloads/)
    2. Download the **macOS installer**.
        ![Python Files](../../assets/guide/environment/3.10.11-files.png "Python Files")
    3. Run the installer and leave all default options enabled.
    4. Verify Python's installation:
        ```bash
        python3 --version
        ```

        ![Verify Python](../../assets/guide/environment/verify-python.png "Verify Python")
    5. Verify `pip`'s installation;
        ```bash
        pip3 --version
        ```

        ![Verify pip](../../assets/guide/environment/verify-pip.png "Verify pip")

=== "Linux"

>     Python is typically preinstalled on most Linux distributions

Reviewer: Worth mentioning common Python-on-Linux pitfalls:
- Don't modify or change the default version of Python that comes with the OS
- `python3` installs whatever the latest version is at the time. How can I (as a user) install an earlier version (maybe I want to use 3.13 instead of 3.14, for whatever reason), or maybe try out a pre-release version?
- How do I handle having multiple versions of Python installed? (a good question for macOS too)

Author: I might just have to not document Linux installs. Yikes 😬

Reviewer: If it makes it easier, we could get away with only mentioning the second point. The first and the third points are perhaps slightly out of scope

Reviewer 2: Uv would simplify the installation process for any platform

    Python is typically preinstalled on most Linux distributions. Keep in mind that you should never modify the system's preinstalled Python environment (using `venv` is always recommended).

    **Debian / Ubuntu**

    ```bash
    sudo apt update
    sudo apt install -y python3 python3-pip
    ```

    **Arch Linux**

    ```bash
    sudo pacman -S python python-pip
    ```

    **Fedora**

    ```bash
    sudo dnf install python3 python3-pip
    ```

    Verify installation:

    ```bash
    python3 --version
    pip3 --version
    ```

## Installing hikari

Once Python and `pip` are installed, installing `hikari` is straightforward. However, you should always make a local environment specific to each project rather than relying on a global environment. Different projects rely on different versions of similar packages, and this prevents compatibility issues in the future.

Open your terminal relative to your project folder and perform the following steps to create a local environment (called a virtual environment or `venv`):

=== "Windows"

    ```powershell
    python -m venv venv
    .\venv\Scripts\activate
    ```

=== "macOS"

    ```bash
    python3 -m venv venv

>     source venv\bin\activate

Reviewer 2: ```suggestion
    source venv/bin/activate
```

    ```

=== "Linux"

    Ensure the `venv` module is installed:
    ```bash
    sudo apt install python3-venv
    ```

    Once installed:
    ```bash
    python3 -m venv venv
    source venv/bin/activate
    ```

Now that the local/virtual environment is created and activated within your project, you may install `hikari`:

=== "Windows"

    ```powershell
    pip install hikari
    ```

=== "macOS"

    ```bash
    pip3 install hikari
    ```

=== "Linux"

    ```bash
    pip3 install hikari
    ```

Verify installation:

```

> python -c "import hikari; print(hikari.__version__)"

Reviewer 2: ```suggestion
python -m hikari
```

this invokes the hikari cli which prints out all of the important information:
```
hikari (2.5.1.dev0) [HEAD]
located at /home/christopher/dev/python/hikari/hikari
CPython 3.14.6 GCC 16.1.1 20260430
Linux Laptop 7.1.5-1-cachyos #1 SMP PREEMPT_DYNAMIC Mon, 27 Jul 2026 14:05:40 +0000 x86_64
```

Reviewer: I would suggest
```suggestion
python -m hikari
```
as the preferred method

Reviewer 2: yes
